Kiali 656 - Adding validations into route rules tab (service details page)

xeviknal opened the pull request:
I added the validations info on top of the route rules list within the Service Details page.

Here is the result:
![validations-weight-table](https://user-images.githubusercontent.com/613814/40166969-1b883b7c-59c0-11e8-992b-7eee307a097a.png)

aljesusg commented on src/pages/ServiceDetails/ServiceDetailsPage.tsx:
```diff
@@ -191,6 +193,16 @@ class ServiceDetails extends React.Component<RouteComponentProps<ServiceId>, Ser
+      .catch(error => {
+        MessageCenter.add(API.getErrorMsg('Could not fetch Service Validations.', error));
+        console.log(error);
+      });
```
Did you forget remove this console.log ?

xeviknal replied:
@aljesusg I did remove them :)

jotak commented:
Just a UX suggestion based on the screenshot: what if you remove the "status" column, and only show an error icon if there's something wrong, next to the weight? In most cases I guess the weights will be correctly set so it's a way not to overload the page with many icons.

Also, maybe the "weights:" label above the tables is redundant as it's repeated in table columns.

xeviknal commented:
@jotak yes, you are right. It might get really overload of red/green/orange icons. Just one thing it comes to my mind is that we might have multiple messages for one line and perhaps one connected to weight and another one connected to the service pointed with the labels.

If you check the [mockups](https://issues.jboss.org/browse/KIALI-555) from serena, she suggested to add this column. Probably, remove the green icons for each correct line would light down the icon overload.

Regarding the weights title. I do agree on removing it.

jotak commented:
@xeviknal I'm looking at the mockups, I didn't notice earlier that the status column is not always present. Does it mean that it's only displayed when there's something relevant to show inside?

xeviknal commented:
@jotak you might be right. Although I remember she added this last column when we were out of time. Perhaps that's why I thought she didn't update the other tables with that column.
@abonas do you remember why we have the status column only for the first scenario?

Anyway, It would seems weird to me to see some tables with 3 columns and other ones with just 2.

For a intermediate scenario, I just managed to make status column thinner and only show icons if there is any validation.

![status-thinner](https://user-images.githubusercontent.com/613814/40131446-51fb648a-593a-11e8-95da-31a66b70292a.png)

aljesusg commented:
For me is fine @xeviknal , about the status column I don't know why @abonas request it. Anyway we can change this in other PR if need it. As you prefer ^^

xeviknal commented:
@lucasponce I agree on the UI consistence so I left the name as it was. In a subsequent PR I will apply this heading as a name in all istio objects tab.

abonas commented:
@aljesusg I did not request the status column. You meant @serenamarie125 , and it is best to ask her directly if there was a mistake in the mockup, instead of everyone here guessing :)

abonas commented:
I think that the table header should be capitalized + bold

abonas commented:
per the mockup it seems that the upper part should contain a notification explaining what is/are the problems. also the upper bar indicator with colors (in the mockup above the table) is missing - is this on purpose?

aljesusg commented:
Sorry @abonas I didn't read correctly ^^

xeviknal commented:
@abonas 

> per the mockup it seems that the upper part should contain a notification explaining what is/are the problems.

That notification is placed next to the route rule name:
![route-rule-notification-wrong](https://user-images.githubusercontent.com/613814/40188715-de9eef72-59fa-11e8-9af6-0409067b8744.png)
When working properly:
![route-rule-notification](https://user-images.githubusercontent.com/613814/40188733-e8975384-59fa-11e8-91a1-57250262bf7e.png)

In the mockups, the route rule name was just over the weights table. It was like that because we didn't consider the fact that route rules (virtual services...) has more configurations than those weights.
Therefore, I moved the weights table next to name and other configs list.

> also the upper bar indicator with colors (in the mockup above the table) is missing - is this on purpose?
Yes, it is missing. That is the bullet char component that is missing in patternfly-react. 
I opened this [JIRA-735](https://issues.jboss.org/browse/KIALI-735) for adding that bullet char once they have it.

xeviknal commented:
@abonas I changed a little bit the layout got get it closer to mockups.

- Now we have the notification in-place right under the route rule name. No need to click on the icon to see that. 
- Also, there is no icon when the route rule doesn't have any broken validations.
- For status column icons, user has to click/hover that icon in order to see the message.

![name-heading-inplace-notification](https://user-images.githubusercontent.com/613814/40228687-3e0fe06e-5a92-11e8-9660-de284cf66ffc.png)

@lucasponce I just changed the heading of each tab also in order to the consistency everywhere, as you mentioned. Here the screenshots:

![deployments-heading](https://user-images.githubusercontent.com/613814/40229566-eb482460-5a94-11e8-9740-ce581e2add9f.png)
![destination-policies](https://user-images.githubusercontent.com/613814/40229567-eb617384-5a94-11e8-8b09-e8131e7fdbe1.png)
![virtual-services-heading](https://user-images.githubusercontent.com/613814/40229568-eb7a7708-5a94-11e8-8308-574a22fd6891.png)
![destination-rules-heading](https://user-images.githubusercontent.com/613814/40229565-eb2faba6-5a94-11e8-936e-7f122967aa91.png)

xeviknal commented:
Are you reviewers agree on merging this one? IMO this is totally mergeable

xeviknal commented:
Thank you! yaaaay! we have iiit! we have it!

On Tue, May 22, 2018 at 1:38 PM, Lucas Ponce <notifications@github.com>
wrote:

> Merged #341 <https://github.com/kiali/kiali-ui/pull/341>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kiali/kiali-ui/pull/341#event-1639079602>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAldtqNHLyJ52cWLjyBH-eGvy5mptYLNks5t0_i8gaJpZM4UBRqL>
> .
>



-- 
Xavier Canal i Masjuan
